core/sr-api/src/lib.rs

```diff
@@ -0,0 +1,589 @@
+		$( #[$attr:meta] )*
+		pub trait $name:ident $(< $( $generic_param_orig:ident $( : $generic_bound_orig:ident )* ),* >)* {
+			$(
+				fn $fn_name:ident $( < $( $fn_generic:ident ),* > )* (
+					$( $param_name:ident : $param_type:ty )*
+				) $( -> $return_ty:ty)*;
+			)*
+		};
+		$( $generic_param_parsed:ident $( : $generic_bound_parsed:path )* ),*;
+		{ $( $result_return_ty:ty; )* };
+		$( $generic_param_joined:ident $( : $generic_bound_joined:ident )*, )*;
+	) => {
+		$( #[$attr] )*
+		pub trait $name < $( $generic_param_parsed $( : $generic_bound_parsed )* ),* > {
+			type Error;
+
+			$(
+				fn $fn_name $( < $( $fn_generic: $crate::Encode + $crate::Decode ),* > )* (
+					&self, at: &$crate::BlockId<Block> $(, $param_name: $param_type )*
+				) -> $result_return_ty;
+			)*
+		}
+	};
+	(@GENERATE_RUNTIME_TRAITS
+		$(
+			$( #[$attr:meta] )*
+			pub trait $name:ident $(< $( $generic_param:ident $( : $generic_bound:ident )* ),* >)* {
+				$(
+					fn $fn_name:ident $( < $( $fn_generic:ident ),* > )* (
+						$( $param_name:ident : $param_type:ty )*
+					) $( -> $return_ty:ty)*;
+				)*
+			};
+		)*
+	) => {
+		decl_apis! {
+			@GENERATE_RUNTIME_TRAITS_WITH_JOINED_GENERICS
+			$(
+				$( #[$attr] )*
+				pub trait $name < $( $( $generic_param $( : $generic_bound )*, )* )* $( $( $( $fn_generic, )* )* )* > {
+					$(
+						fn $fn_name ($( $param_name: $param_type ),*) $( -> $return_ty )*;
+					)*
+				}
+			)*
+		}
+	};
+	(@GENERATE_RUNTIME_TRAITS_WITH_JOINED_GENERICS
+		$(
+			$( #[$attr:meta] )*
+			pub trait $name:ident < $( $generic_param:ident $( : $generic_bound:ident )*, )* > {
+				$(
+					fn $fn_name:ident($( $param_name:ident : $param_type:ty )*) $( -> $return_ty:ty)*;
+				)*
+			}
+		)*
+	) => {
+		pub mod runtime {
+			use super::*;
+
+			$(
+				$( #[$attr] )*
+				pub trait $name < $( $generic_param $( : $generic_bound )* ),* > {
+					$(
+						fn $fn_name ($( $param_name: $param_type ),*) $( -> $return_ty )*;
+					)*
+				}
+			)*
+		}
+	};
+}
+
+decl_apis! {
+	/// The `Core` api trait that is mandantory for each runtime.
+	pub trait Core<Block: BlockT, AuthorityId> {
+		fn version() -> RuntimeVersion;
+		fn authorities() -> Vec<AuthorityId>;
+		fn execute_block(block: Block);
+	}
+
+	/// The `Metadata` api trait that returns metadata for the runtime.
+	pub trait Metadata {
+		fn metadata() -> Vec<u8>;
+	}
+
+	/// The `BlockBuilder` api trait that provides required functions for building a block for a runtime.
+	pub trait BlockBuilder<Block: BlockT> {
+		fn initialise_block(header: <Block as BlockT>::Header);
+		fn apply_extrinsic(extrinsic: <Block as BlockT>::Extrinsic) -> ApplyResult;
+		fn finalise_block() -> <Block as BlockT>::Header;
+		fn inherent_extrinsics<InherentExtrinsic, UncheckedExtrinsic>(inherent: InherentExtrinsic) -> Vec<UncheckedExtrinsic>;
+		fn random_seed() -> <Block as BlockT>::Hash;
+	}
+
+	/// The `OldTxQueue` api trait for interfering with the old transaction queue.
+	pub trait OldTxQueue {
```

Comment:
again, prefer to see this stuff not so central since it's not actually "core" to substrate.